src/common/common_init.cc

```diff
@@ -12,6 +12,8 @@
  *
  */
 
+#include "include/compat.h"
+#include <sys/stat.h>
```

Comment:
why you are adding `<sys/stat.h>` ? if you are expecting `S_IRWXU` and its friend to be visible, you might want to `#include <sys/stat.h>` in `include/compat.h` instead.

Reply:
@tchaikov 
I contemplated about doing that, but then again `compat.h` gets loaded with all kinds of includes.
Argument for doing so, is that it is already done for quite a few more include files.

Reply:
yeah, i sorta feel the same. if we want to avoid bloating `compat.h`, probably we can move the definition of `ACCESSPERMS` back to where it's used. the downside of this approach is that we are repeating.

but i don't think it hurts that much..

Reply:
@tchaikov 
For now I included it in `compat.h`.
There are some remarks in the coding style suggesting that a include file should fix its own deps.
So be it.